tests/components/compensation/test_sensor.py

```diff
@@ -1,115 +1,209 @@
 """The tests for the integration sensor platform."""
 
+from typing import Any
+from unittest.mock import patch
+
 import pytest
 
+from homeassistant import config as hass_config
 from homeassistant.components.compensation.const import CONF_PRECISION, DOMAIN
 from homeassistant.components.compensation.sensor import ATTR_COEFFICIENTS
-from homeassistant.components.sensor import DOMAIN as SENSOR_DOMAIN
+from homeassistant.components.sensor import (
+    ATTR_STATE_CLASS,
+    SensorDeviceClass,
+    SensorStateClass,
+)
 from homeassistant.const import (
+    ATTR_DEVICE_CLASS,
     ATTR_UNIT_OF_MEASUREMENT,
     EVENT_HOMEASSISTANT_START,
     EVENT_STATE_CHANGED,
+    SERVICE_RELOAD,
+    STATE_UNAVAILABLE,
     STATE_UNKNOWN,
+    UnitOfTemperature,
 )
 from homeassistant.core import HomeAssistant
 from homeassistant.setup import async_setup_component
 
+from tests.common import assert_setup_component, get_fixture_path
+
+TEST_OBJECT_ID = "test_compensation"
+TEST_ENTITY_ID = "sensor.test_compensation"
+TEST_SOURCE = "sensor.uncompensated"
+
+TEST_BASE_CONFIG = {
+    "source": TEST_SOURCE,
+    "data_points": [
+        [1.0, 2.0],
+        [2.0, 3.0],
+    ],
+    "precision": 2,
+}
+TEST_CONFIG = {
+    "name": TEST_OBJECT_ID,
+    "unit_of_measurement": "a",
+    **TEST_BASE_CONFIG,
+}
 
-async def test_linear_state(hass: HomeAssistant) -> None:
-    """Test compensation sensor state."""
-    config = {
-        "compensation": {
-            "test": {
-                "source": "sensor.uncompensated",
-                "data_points": [
-                    [1.0, 2.0],
-                    [2.0, 3.0],
-                ],
-                "precision": 2,
-                "unit_of_measurement": "a",
-            }
-        }
-    }
-    expected_entity_id = "sensor.compensation_sensor_uncompensated"
 
-    assert await async_setup_component(hass, DOMAIN, config)
-    assert await async_setup_component(hass, SENSOR_DOMAIN, config)
+async def async_setup_compensation(
+    hass: HomeAssistant, count: int, config: dict[str, Any]
+) -> None:
+    """Do setup of a compensation integration sensor."""
+    with assert_setup_component(count, DOMAIN):
```

Comment:
As `count` is 1 in each test, I suggest hardcoding it for now. This removes the need to specify it everywhere.
If in the future we really need to change it, we can refactor it there

```suggestion
    with assert_setup_component(1, DOMAIN):
```